glyph-brush/src/glyph_brush.rs

```diff
@@ -599,40 +597,40 @@ impl SectionHashDetail {
         let mut s = build_hasher.build_hasher();
         layout.hash(&mut s);
         parts.hash_text_no_color(&mut s);
-        let text_no_color_alpha_hash = s.finish();
-
-        parts.hash_alpha(&mut s);
-        let text_no_color_hash = s.finish();
+        let text_hash = s.finish();
 
         parts.hash_color(&mut s);
-        let text_hash = s.finish();
+        let text_color_hash = s.finish();
+
+        parts.hash_alpha(&mut s);
+        let test_alpha_color_hash = s.finish();
 
         parts.hash_geometry(&mut s);
         parts.hash_z(&mut s);
         let full_hash = s.finish();
 
         Self {
-            text_no_color_alpha: text_no_color_alpha_hash,
-            text_no_color: text_no_color_hash,
             text: text_hash,
-            // text_geometry: text_geo_hash,
+            text_color: text_color_hash,
+            test_alpha_color: test_alpha_color_hash,
             full: full_hash,
             geometry: SectionGeometry::from(section),
         }
     }
 
     /// They're different, but how?
-    fn diff(self, other: SectionHashDetail) -> SectionHashDiff {
-        if self.text == other.text {
-            return SectionHashDiff::GeometryChange;
+    fn diff(self, other: SectionHashDetail) -> GlyphChange {
+        if self.test_alpha_color == other.test_alpha_color {
+            return GlyphChange::Geometry(self.geometry);
         } else if self.geometry == other.geometry {
-            if self.text_no_color == other.text_no_color {
-                return SectionHashDiff::ColorChange;
-            } else if self.text_no_color_alpha == other.text_no_color_alpha {
-                return SectionHashDiff::AlphaChange;
```

Comment:
The issue was here, and with my dodgy naming making it hard (for me) to think about this.

Here it says `text+alpha is different && text is same ==> only alpha has changed` which is wrong as the color **and** alpha may have changed.